client/src/components/PageLayout/Header.tsx

```diff
@@ -40,13 +102,43 @@ const Header: React.FC<{ classes: Record<string, any> }> = ({ classes }) => {
       </Link>
       <Grid component="nav" item xs={12} md={4}>
         <Grid container direction="row" spacing={2} justify="center">
-          {headerLinks.map(headerLink => (
-            <Grid item key={headerLink.name}>
-              <Link href={headerLink.href}>
-                <a className={styles.link}>{headerLink.label}</a>
-              </Link>
-            </Grid>
-          ))}
+          {headerLinks.map(headerLink => {
+            if (headerLink.name === 'google' && !isAuth) {
+              return (
+                <Grid item key={headerLink.name}>
+                  <a className={styles.link} href={headerLink.href}>
+                    {headerLink.label}
+                  </a>
+                </Grid>
+              );
+            } else if (headerLink.name === 'google' && isAuth) {
+              return null;
+            }
+
+            if (headerLink.name === 'logout' && isAuth) {
+              return (
+                <Grid item key={headerLink.name}>
+                  <a className={styles.link} href={headerLink.href}>
+                    {headerLink.label}
+                  </a>
+                </Grid>
+              );
+            } else if (headerLink.name === 'logout' && !isAuth) {
+              return null;
+            }
+
+            if (headerLink.name === 'login' && isAuth) {
+              return null;
+            }
```

Comment:
This can probably be rewritten to be easier to understand - maybe something like the following, outside of the .map call:

```js
const links = headerLinks;

if (isAuth) {
  links.push(logout button config object);
} else {
  links.push(login button config object);
}

links.map(link => etc etc
```